extensions/dedup_messages.go

```diff
@@ -127,26 +109,36 @@ func (d dedup) IsUnique(ctx context.Context, device, msg, game, platform string)
 		}
 	}
 
-	log.Debug("Deduplication sampling percentage check passed, proceeding with deduplication check")
 	rdbKey := keyFor(device, msg)
 
-	// Store the key in Redis with a placeholder value ("1")—the actual value is irrelevant, as we only need to check for key existence.
-	unique, err := d.redis.SetNX(ctx, rdbKey, "1", d.ttl).Result()
+	rArgs := &redis.SetArgs{
+		Mode: `NX`, 
+		TTL:  d.ttl,
+	}
 
+	// Store the key in Redis with a placeholder value ("1")—the actual value is irrelevant, as we only need to check for key existence.
+	unique, err := d.redis.SetArgs(ctx, rdbKey, "1", *rArgs).Result();
+	
 	if err != nil {
-		log.WithFields(logrus.Fields{
-			"error":  err,
-			"key":    rdbKey,
-			"game":   game,
-		}).Error("Failed to check message dedup in Redis")
-
-		StatsReporterDedupFailed(d.statsReporters, game, platform)
+		if errors.Is(err, redis.Nil) {
+			log.WithField("key", rdbKey).Debug("Message is not unique, key already exists in Redis")
+			return false
+
+		} else {
+			log.WithFields(logrus.Fields{
```

Comment:
don't this log have something like `WithError(err)`? Then you get the stack pretty-printed